client/signup/steps/p2-site/index.jsx

```diff
@@ -165,12 +183,45 @@ class P2Site extends React.Component {
 
 							this.logValidationErrorToLogstash( error.error, errorMessage );
 						} else {
-							messages.site = {
-								[ error.error ]: error.message,
-							};
-
+							if ( SITE_TAKEN_ERROR_CODES.includes( error.error ) ) {
+								messages.site = {
+									[ ERROR_CODE_TAKEN_SITE ]: this.props.translate(
+										'Sorry, that site already exists! Here are some available alternatives:'
+									),
+								};
+							} else {
+								messages.site = {
+									[ error.error ]: error.message,
+								};
+							}
+
+							// We want to log the real error code and message. The above is formatted for the end user
+							// only.
 							this.logValidationErrorToLogstash( error.error, error.message );
 						}
+
+						if ( error.error && SITE_TAKEN_ERROR_CODES.includes( error.error ) ) {
+							WPCOM_SUBDOMAIN_SUFFIX_SUGGESTIONS.forEach( ( suffix ) => {
+								const suggestedSubdomain = `${ fields.site }${ suffix }.wordpress.com`;
+
+								wpcom
+									.domains()
+									.suggestions( {
+										quantity: 1,
+										query: suggestedSubdomain,
```

Comment:
You can (and probably should) simply submit `fields.site suffix` - no need to add `.wordpress.com`. The `only_wordpressdotcom` parameter is what forces all the responses to have `.wordpress.com` suffixes.